src/components/KButton/KButton.vue

```diff
@@ -110,6 +130,25 @@ const strippedAttrs = computed((): typeof attrs => {
   return modifiedAttrs
 })
 
+const stop = (event: Event) => {
+  event.preventDefault()
+  event.stopPropagation()
+}
+
+const listeners = computed(() => {
+  if (!props.disabled || buttonType.value === 'button') {
+    return {}
+  }
+
+  // Only prevent clicks on disabled links, emulating the same behavior as a disabled button
+  return {
+    clickCapture: stop,
+    dblclickCapture: stop,
+    mousedownCapture: stop,
+    mouseupCapture: stop,
+  }
+})
```

Comment:
I can still right-click on a disabled button in the sandbox with a `to` prop, is that intended?

![image](https://github.com/user-attachments/assets/18132b57-38a2-418a-a506-b52d44be06b7)

Reply:
If it's simply a button (e.g. with a `@click` handler) I think just preventing the click as you have it is sufficient.

For links, I agree, just removing the link itself may be more ideal; however, at that point rendering an `<a>` tag without the `href` is not semantically correct.

Instead, we could do something like this?

```html
<a 
  href="https://example.com" 
  aria-disabled="true" 
  @click.prevent="">Disabled Link</a>
```

Reply:
> Instead, we could do something like this?
> 
> ```
> <a 
>   href="https://example.com" 
>   aria-disabled="true" 
>   @click.prevent="">Disabled Link</a>
> ```

This leaves the URL visible, I think we should not expose it to user if it's unavailable:

<img width="272" alt="image" src="https://github.com/user-attachments/assets/7158cf0a-d937-4d2c-9a90-266442468d5f">

Or shall we just follow ReactAria's approach? They seem to have put tons of efforts on A11Y.

Reply:
I don't think "exposing" the URL is an issue, it seems like the use-case is just not allowing them click it. If it was a security concern, you'd remove it from the DOM

Reply:
`<a role="link">` seem to be fine (at lease in Chrome's Accessibility Tree):

<img width="266" alt="image" src="https://github.com/user-attachments/assets/4362782b-e88f-4f51-b949-0512e7ea764a">

It's rendered the same way as `<a href>` in the a11y tree.

`<a href>` should be equal to `[role="link"]` as described [here](https://www.w3.org/TR/html-aria/#dfn-naming-prohibited:~:text=href-,role%3Dlink,-Roles%3A%20button).

Reply:
Done. Need a final review. @adamdehaven